src/Symfony/Component/Security/Core/Tests/Authentication/Token/AbstractTokenTest.php

```diff
@@ -96,6 +97,19 @@ public function testSerialize()
         $this->assertEquals($token->getAttributes(), $uToken->getAttributes());
     }
 
+    public function testSerializeWithRoleObjects()
+    {
+        $user = new User('name', 'password', array(new Role('ROLE_FOO'), new Role('ROLE_BAR')));
+        $token = new ConcreteToken($user, $user->getRoles());
+
+        $serialized = serialize($token);
+        $unserialized = unserialize($serialized);
+
+        $roles = $unserialized->getRoles();
+
+        $this->assertEquals($roles, $user->getRoles());
```

Comment:
Same test as above, ie. `$this->assertEquals($token->getRoles(), $uToken->getRoles());`